boot/bootutil/src/boot_record.c

```diff
@@ -346,7 +353,28 @@ int boot_save_shared_data(const struct image_header *hdr, const struct flash_are
     }
 #endif
 
+#if !defined(MCUBOOT_SINGLE_APPLICATION_SLOT) && defined(MCUBOOT_HW_ROLLBACK_PROT)
+    image = 0;
+    while (image < BOOT_IMAGE_NUMBER && !rc) {
+        FIH_CALL(boot_nv_security_counter_get, fih_rc, image, &security_cnt);
+        if (FIH_NOT_EQ(fih_rc, FIH_SUCCESS)) {
+            /* Some platforms support only a single security counter. */
+            continue;
+        }
+
+        rc = boot_add_data_to_shared_area(TLV_MAJOR_BLINFO,
+                                          BLINFO_SECURITY_COUNTER_IMAGE_0 + image,
+                                          sizeof(security_cnt),
+                                          (void *)&security_cnt);
+        if (!rc) {
+            break;
```

Comment:
I see this is not present in the loop below. Perhaps similar code should be also added there?

Reply:
@ahasztag Can you re-review?